app/lib/test_executor.rb

```diff
@@ -16,9 +16,8 @@ def self.correct_submission?(submission)
 
   def self.correct_output?(submission, test)
     output = run_test(submission, test)
-    if test.expected_output.nil?
-      fill_expected_output(submission.problem, test)
-    end
+
+    raise "Missing expected output" if test.expected_output.nil?
```

Comment:
A this point, the test must have an expected output. If there is not one, it's a bug that we need to investigate